include/nvs/nvs.h

```diff
@@ -0,0 +1,185 @@
+/*  NVS: non volatile storage in flash
+ *
+ * Copyright (c) 2018 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __NVS_H_
+#define __NVS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <sys/types.h>
+/**
+ * @brief Non-volatile Storage
+ * @defgroup nvs Non-volatile Storage
+ * @{
+ * @}
+ */
+
+/**
+ * @brief Non-volatile Storage Data Structures
+ * @defgroup nvs_data_structures Non-volatile Storage Data Structures
+ * @ingroup nvs
+ * @{
+ */
+
+/**
+ * @brief Non-volatile Storage File system structure
+ *
+ * @param magic File system magic, repeated at start of each sector
+ * @param write_location Next write location for entry header
+ * @param offset File system offset in flash
+ * @param sector_id Counter for sector
+ * @param sector_size File system is divided into sectors each sector should be
+ * multiple of pagesize and also a power of 2
+ * @param free_space Indicator for available space in the filesystem
+ * @param max_len Maximum size of storage items
+ * @param sector_count Amount of sectors in the file systems
+ * @param entry_sector Oldest sector in used
+ * @param write_block_size Alignment size in bytes_to_copy
+ * @param nvs_lock Mutex
+ * @param flash_device Flash Device
+ */
+struct nvs_fs {
+	u32_t magic; /* filesystem magic, repeated at start of each sector */
+	off_t write_location; /* next write location for entry header */
+	off_t offset; /* filesystem offset in flash */
+	u16_t sector_id; /* sector id, a counter for each created sector */
+	u16_t sector_size; /* filesystem is divided into sectors,
+			    * sector size should be multiple of pagesize
+			    * and a power of 2
+			    */
+	u16_t free_space; /* Indicator for available space in the file system.
+			   * Writes are only allowed if free_space is equal to
```

Comment:
Writes are only allowed if free_space is greater or equal to ...

Reply:
I deducted this form https://github.com/zephyrproject-rtos/zephyr/pull/6391/files#diff-84f8a9960aa570dae836d5cb80c9f46aR707
but I see - here is:
https://github.com/zephyrproject-rtos/zephyr/pull/6391/files#diff-84f8a9960aa570dae836d5cb80c9f46aR776